squidDrone/src/com/com_message_buffer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef SRC_COM_MESSAGE_BUFFER_HPP_
+#define SRC_COM_MESSAGE_BUFFER_HPP_
+
+#include <array>
+#include <queue>
+#include "com_types.hpp"
+
+namespace com {
+  /** 
+    * @brief This class represents a queue style buffer for communication
+    * frames exchanged between drones.
+    * 
+    */
+  class ComMessageBuffer {
+    public:
+      /** 
+       * @brief Construct a new Com Message Buffer object
+       * 
+       */
+      ComMessageBuffer(){};
+
+      /** 
+       * @brief Transfer data to queue buffer. 
+       * 
+       * @param data Reference to data frame. Must conform to data frame specifications.
+       * @return uint8_t Returns 0 on no error, otherwise negative value.
+       * 
+       */
+      types::ComError put_data(std::array<std::uint8_t, 32> &data);
+
+      /** 
+       * @brief Retrieve data frame from queue buffer.
+       * 
+       * @return std::array<std::uint8_t, 32> 
+       * 
+       */
+      std::array<std::uint8_t, 32> get_data();
```

Comment:
why did you switch from trailing return type style to plain old return type style? Do we want to only use one style, or mix both?

Reply:
Only using trailing return types now.